builder/evaluator.go

```diff
@@ -168,6 +171,9 @@ func (b *builder) Run(context io.Reader) (string, error) {
 	// some initializations that would not have been supplied by the caller.
 	b.Config = &runconfig.Config{}
 
+	// initialize the allowed args
+	b.Config.AllowedBuildArgs = make(map[string][]string)
```

Comment:
This appears in the image metadata as:
```
        "AllowedBuildArgs": {
            "bar": [
                "--default=xxx"
            ],
            "foo": [
                "--default=yyy",
                "--description=hello there"
            ]
        }
```
What do you think about switching this so that it is strong typed?
```
        "AllowedBuildArgs": {
            "bar": {
                "default": "xxx"
            },
            "foo": {
                "default": "yyy",
                "description": "hello there"
            }
        }
```
or will that be too much of pain?